bindings_ffi/src/mls.rs

```diff
@@ -1023,22 +1008,16 @@ impl FfiGroup {
         Ok(())
     }
 
-    pub async fn stream(
-        &self,
-        message_callback: Box<dyn FfiMessageCallback>,
-    ) -> Result<Arc<FfiStreamCloser>, GenericError> {
+    pub fn stream(&self, message_callback: Box<dyn FfiMessageCallback>) -> FfiStreamCloser {
         let inner_client = Arc::clone(&self.inner_client);
-        let stream_closer = MlsGroup::stream_with_callback(
+        let handle = MlsGroup::stream_with_callback(
             inner_client,
             self.group_id.clone(),
             self.created_at_ns,
             move |message| message_callback.on_message(message.into()),
-        )?;
+        );
 
-        Ok(Arc::new(FfiStreamCloser {
-            close_fn: stream_closer.close_fn,
-            is_closed_atomic: stream_closer.is_closed_atomic,
-        }))
+        FfiStreamCloser::new(handle)
```

Comment:
Why do we hold the error inside of the `FfiStreamCloser` instead of returning it to the caller of this method?

Reply:
The stream is started inside of the `tokio::spawn`, so the handle is returning the error once the stream is `Joined`/awaited. 

I think a good followup is [this](https://github.com/xmtp/libxmtp/issues/899) to allow the sdks to handle the errors as necessary